src/Listener/CodePatchListener.php

```diff
@@ -36,34 +38,40 @@ public function __construct(CodePatcher $codePatcher)
      */
     public function patch(ExerciseRunnerEvent $event): void
     {
-        $fileName = $event->getInput()->getArgument('program');
+        $files = [
+            $event->getInput()->getArgument('program'),
+        ];
 
-        if (null === $fileName) {
-            return;
+        $exercise = $event->getExercise();
+        if ($exercise instanceof ProvidesSolution) {
+            $files[] = $exercise->getSolution()->getEntryPoint();
         }
 
-        $this->originalCode = (string) file_get_contents($fileName);
-        file_put_contents(
-            $fileName,
-            $this->codePatcher->patch($event->getExercise(), $this->originalCode)
-        );
+        foreach ($files as $fileName) {
+            if (null === $fileName) {
+                return;
+            }
+
+            $this->originalCode[$fileName] = (string) file_get_contents($fileName);
+
+            file_put_contents(
+                $fileName,
+                $this->codePatcher->patch($event->getExercise(), $this->originalCode[$fileName])
+            );
+        }
     }
 
     /**
-     * @param ExerciseRunnerEvent $event
+     * @param \PhpSchool\PhpWorkshop\Event\EventInterface $event
      */
-    public function revert(ExerciseRunnerEvent $event): void
+    public function revert(\PhpSchool\PhpWorkshop\Event\EventInterface $event): void
```

Comment:
Any event can revert, we're not requiring anything from the solution here so felt ok to "open" it up in this regard. Allows for tear down / exception events dispatched to come here and re-use the `originalCode` data

Reply:
Yeah sounds legit 👍